sympy/solvers/tests/test_ode.py

```diff
@@ -3830,3 +3830,15 @@ def test_issue_15996():
     sol = Eq(f(x), C1 + (C2 + C3*x - x**2/8 + 5*exp(2*I*x)/16)*sin(x) + (C4 + C5*x + I*x**2/8 + 5*I*exp(2*I*x)/16)*cos(x) - I*exp(I*x))
     assert sol == dsolve(eq, hint='nth_linear_constant_coeff_variation_of_parameters')
     assert checkodesol(eq, sol) == (True, 0)
+
+
+def test_issue_18408():
+    eq = f(x).diff(x, 3) - f(x).diff(x) - sinh(x)
+    sol = Eq(f(x), C1 + C2*exp(-x) + C3*exp(x) + x*sinh(x)/2)
```

Comment:
It would be nice to have a solution in terms of sinh and cosh in this case, but that might not be straightforward to implement.

Reply:
I've often thought the same about `exp(I*x)` in some cases that would be preferred rather than `cos/sin` e.g.:
```julia
In [15]: dsolve(f(x).diff(x, 2) + f(x) - exp(I*x))                                                                                             
Out[15]: 
                                    ⅈ⋅x
                               ⅈ⋅x⋅ℯ   
f(x) = C₁⋅sin(x) + C₂⋅cos(x) - ────────
                                  2  
```
Maybe there should be a parameter to dsolve to influence which type of solutions it generates.

Reply:
I think it could at least be smarter based on what sorts of functions are present in the ODE itself. But this can be done in a future PR.